dht.go

```diff
@@ -67,8 +66,8 @@ type IpfsDHT struct {
 }
 
 // NewDHT creates a new DHT object with the given peer as the 'local' host
-func NewDHT(ctx context.Context, h host.Host, dstore ds.Batching) *IpfsDHT {
-	dht := NewDHTClient(ctx, h, dstore)
+func NewDHT(ctx context.Context, h host.Host, dstore ds.Batching, validator record.Validator) *IpfsDHT {
```

Comment:
I am not sure I like having a required record validator argument; it will also flat out break existing code.
Can we make this optional with a variadic?

Reply:
The default should be what the old default functionality was, public key validation so that it's backwards compatible.

Reply:
also, ergonomics.

Reply:
Yeah, good point. I was in a "ftfy" mood when I made this change.

What about just going back to exposing a Validator public field. That's closer to what we currently do (although racy if the host is up and accepting streams). The nice alternative is an options system (more code but probably better in the long run).

Reply:
let's make it nice, there is no rush.